resolwe/flow/executors/docker/__init__.py

```diff
@@ -30,15 +36,56 @@ def start(self):
             'container_image': self.requirements.get('image', settings.FLOW_EXECUTOR['CONTAINER_IMAGE']),
         }
 
+        # Set resource limits.
+        limits = []
+        if 'cores' in self.resources:
+            # Each core is equivalent to 1024 CPU shares. The default for Docker containers
+            # is 1024 shares (we don't need to explicitly set that).
+            limits.append('--cpu-shares={}'.format(int(self.resources['cores']) * 1024))
```

Comment:
I agree that it would be useful to expose all requirements to processes so they can, for example, bound their memory usage to the given resource limit.

Regarding the implementation, I'm more in favor of just making `requirements` an invalid input name 🙂 . I think it will be easier for process writers to use `requirements.resources.memory` or `requirements.resources.cores` than `_requirements.resources.memory` or `_requirements.resources.cores`.